app/scripts/lib/network-store.js

```diff
@@ -50,16 +50,24 @@ export default class ReadOnlyNetworkStore {
     return this._state;
   }
 
+  /**
+   * Set metadata/version state
+   *
+   * @param {object} metadata - The metadata/version data to set
+   */
+  setMetadata(metadata) {
+    this._metadata = metadata;
+  }
+
   /**
    * Set state
    *
    * @param {object} state - The state to set
-   * @returns {Promise<void>}
    */
   async set(state) {
     if (!this._initialized) {
       await this._initializing;
     }
-    this._state = state;
+    this._state = { data: state, meta: this._metadata };
```

Comment:
Nit: This should have similar validation, so that our e2e tests can catch cases where we try to persist at the wrong time

Reply:
done [here](https://github.com/MetaMask/metamask-extension/pull/15978/commits/93229e356ba6fff173b386651e4287d6d99b930f)